sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusSharedKeyCredential.java

```diff
@@ -165,6 +152,18 @@ private AccessToken generateSharedAccessSignature(final String resource) throws
             return new AccessToken(sharedAccessSignature, getExpirationTime(sharedAccessSignature));
         }
 
+        final Mac hmac;
+        try {
+            hmac = Mac.getInstance(HASH_ALGORITHM);
+            hmac.init(secretKeySpec);
+        } catch (NoSuchAlgorithmException e) {
+            throw LOGGER.logExceptionAsError(new UnsupportedOperationException(
+                String.format("Unable to create hashing algorithm '%s'", HASH_ALGORITHM), e));
```

Comment:
nit: Since `HASH_ALGORITHM` is a constant, do we need to String.format on each error? (super minutia here as if this fails it will always fail, and maybe we should do something special for that to prevent retries)

Reply:
Thanks, changed error string to static.

Reg retry: For low-level clients, non-transient errors are thrown to the application. For high-level clients, such errors are sent to the application error handler. The high-level layer will then attempt to recover by recreating the low-level client. This is by design for Processor that it won't filter or further examine non-transient errors, some infra related non-transient errors get recovered (like missing entity gets (re)created from somewhere else), but idea is Processor runs for ever